content/frameworks/react/virtualize-long-lists-react-window/index.md

```diff
@@ -0,0 +1,286 @@
+---
+page_type: guide
+title: Virtualize large lists with react-window
+description: |
+  `react-window` is a library that allows large lists to be rendered efficiently.
+author: houssein
+web_lighthouse: N/A
+web_updated_on: N/A # TODO: update
+web_published_on: N/A # TODO: update
+wf_blink_components: N/A
+---
+
+# Virtualize long lists with react-window
+
+`react-window` is a library that allows large lists to be rendered efficiently.
+
+Here's an example of a list that contains 1000 rows being rendered with `react-window`. Try scrolling as fast you can.
+
+<div class="glitch-embed-wrap" style="height: 750px; width: 100%;">
+  <iframe
+    src="https://glitch.com/embed/#!/embed/react-window?path=src/App.js&previewSize=100"
+    alt="react-window on Glitch"
+    style="height: 100%; width: 100%; border: 0;">
+  </iframe>
+</div>
+
+## Why is this useful?
+
+There may be times where you need to display a large table or list that contains many rows. Loading every single item on a massive list can affect performance significantly.
+
+**List virtualization**, or "windowing", is the concept of only rendering what is visible to the user. The number of elements that are rendered at first is a very small subset of the entire list and the "window" of visible content _moves_ when the user continues to scroll. This improves both the rendering and scrolling performance of the list. 
+
+![Window diagram](./window-diagram.png)
+
+DOM nodes that exit the "window" are recycled, or immediately replaced with newer elements as the user scrolls down the list. This keeps the number of all rendered elements specific to the size of the window.
+
+![DOM nodes changing on scroll](./scroll.gif)
+
+## react-window
+
+**`react-window`** is a third-party library that makes it easier to create virtualized lists in your application. It provides a number of base APIs that can be used for different types of lists and tables.
+
+`react-window` is a rewrite of `react-virtualized`, a different windowing library built by the same author. It's a smaller and faster alternative with APIs and docs that are subjectively easier to use.
+
+<blockquote cite="https://github.com/bvaughn/react-window">
+    <p><i>"Adding a react-virtualized list to a CRA project increases the (gzipped) build size by ~33.5 KB. Adding a react-window list to a CRA project increases the (gzipped) build size by &lt;2 KB."</i></p>
+    <p>
+      <cite>From <a href="https://github.com/bvaughn/react-window">react-window GitHub</a></cite>.
+    </p>
+</blockquote>
+
+### FixedSizeList
```

Comment:
Suggestion to re-title the components to make it very clear what's being covered. Something like "When to use FixedSizeList component"

Reply:
Changed to `When to use fixed size lists`